cellprofiler/modules/measuretexture.py

```diff
@@ -124,10 +125,28 @@
 IO_BOTH = "Both"
 
 
+def haralick_2d(image, distances, levels, ignore_zeros=True):
+    angles = [
+        0,
+        1 * numpy.pi / 4,
+        1 * numpy.pi / 2,
+        3 * numpy.pi / 4
+    ]
+
+    greycomatrix = skimage.feature.greycomatrix(image, distances, angles, levels, symmetric=True)
```

Comment:
Although it was previously suggested, I wasn't entirely certain if it should be done. But, if it's the desired behavior, I'll be happy to add it in. Should a loss of precision warning be issued like in [scikit-image](https://github.com/scikit-image/scikit-image/blob/master/skimage/util/dtype.py#L134) line 134 `warn("Possible precision loss when converting from {} to {}"`?

Reply:
The documentation for greycomatrix:

`levelsint, optional`

`The input image should contain integers in [0, levels-1], where levels indicate the number of grey-levels counted (typically 256 for an 8-bit image). This argument is required for 16-bit images or higher and is typically the maximum of the image. As the output matrix is at least levels x levels, it might be preferable to use binning of the input image rather than large values for levels.`

So yeah, I think we need to set it as integers 0-[levels-1].

Reply:
created PR for this comment: https://github.com/bethac07/CellProfiler/pull/11